app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt

```diff
@@ -106,29 +151,169 @@ class PlatformParameterIntegrationTest {
-    platformParameterController.updatePlatformParameterDatabase(mockPlatformParameterList)
+  fun testIntegration_updateEmptyDatabase_readDatabase_checkWelcomeMsgIsVisible() {
+    platformParameterController.updatePlatformParameterDatabase(
+      mockPlatformParameterListWithToastEnabled
+    )
     testCoroutineDispatchers.runCurrent()
 
-    launch(SplashTestActivity::class.java).use {
+    launch(SplashTestActivity::class.java).use { scenario ->
+      // Fetch the latest platform parameter from cache store after execution of work request to
+      // imitate the loading process at the start of splash test activity.
+      scenario.onActivity { activity ->
+        activity.splashTestActivityPresenter.loadPlatformParameters()
+      }
       testCoroutineDispatchers.runCurrent()
+
+      assertThat(ShadowToast.getLatestToast()).isNotNull()
+      assertThat(ShadowToast.getTextOfLatestToast()).isEqualTo(SplashTestActivity.WELCOME_MSG)
+    }
+  }
+
+  @Test
+  fun testIntegration_executeSyncUpWorkerCorrectly_readDatabase_checkWelcomeMsgIsVisible() {
+    launch(SplashTestActivity::class.java).use { scenario ->
+      // Set up versionName to get correct network response from mock platform parameter service.
+      setUpApplicationForVersionName(MockPlatformParameterService.appVersionForCorrectResponse)
+      platformParameterController.updatePlatformParameterDatabase(
+        mockPlatformParameterListWithToastDisabled
+      )
+
+      val workManager = WorkManager.getInstance(context)
+      val requestId = setUpAndEnqueueSyncUpWorkerRequest(workManager)
+      testCoroutineDispatchers.runCurrent()
+
+      val workInfo = workManager.getWorkInfoById(requestId)
+      // Check the work request succeeded which means the local database was updated with new values.
+      assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
+
+      // Fetch the latest platform parameter from cache store after execution of work request to
+      // imitate the loading process at the start of splash test activity.
+      scenario.onActivity { activity ->
+        activity.splashTestActivityPresenter.loadPlatformParameters()
+      }
+      testCoroutineDispatchers.runCurrent()
+
+      // As the local database was updated correctly the app will use the server values, and the
+      // server value for the splash screen welcome msg param is true.
       assertThat(ShadowToast.getLatestToast()).isNotNull()
       assertThat(ShadowToast.getTextOfLatestToast()).isEqualTo(SplashTestActivity.WELCOME_MSG)
     }
   }
 
+  @Test
+  fun testIntegration_executeSyncUpWorkerIncorrectly_readDatabase_checkWelcomeMsgIsInvisible() {
+    launch(SplashTestActivity::class.java).use { scenario ->
+      // Set up versionName to get incorrect network response from mock platform parameter service.
+      setUpApplicationForVersionName(MockPlatformParameterService.appVersionForWrongResponse)
```

Comment:
This flow seems a bit odd; shouldn't the app be set up before launching the test activity?

Reply:
We are doing so because we needed to change the app version for testing the work request for different network response. But if we were doing so before the activity is launched then the SplashTestActivity was not able to be launched by tests. @anandwana001, @vinitamurthi  and @jcqli are known of this issue but thre is no such fix for this as based on our research. Also even if we are doing it this way, we actually fetch the values after the work request is completed. Therefore it still checks the integration correctly

Reply:
I'd need more info on what specifically is preventing us from doing the other approach, but given this is a test arrangement I'm not too bothered by this.